src/deluge/dsp/oscillators/sine_osc.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include "deluge/util/misc.h"
+#include <array>
+#include <cstdint>
+
+namespace deluge::dsp {
+class SineOsc {
+	static constexpr size_t kSineTableSizeMagnitude = 8;
+	static constexpr std::array<int16_t, 512> sineWaveDiff __attribute__((aligned(CACHE_LINE_SIZE))) = {
```

Comment:
I think we've had problems with the compiler duplicating initializations like this when they're in headers in the past, so this might need to stay in lookuptables.cpp.

Reply:
mmm. goodpoint. worst comes to worst I can move it to the cpp file.

Reply:
I'm not a fan of having them all grouped up like that tho

Reply:
moved the table to the implementation file